### ISSUE TITLE
Postgres Socket "Bad Address"

### ISSUE THREAD

**Issue by Retr0astic** [cioraneanu/firefly-pico#126]
Hi, 
The latest version, version 1.4.2 gives me "nc: bad address '/var/run/postgresql'" on container start, only this version has this error on the stable branch.

I connect to postgres through socket.

**Comment by cioraneanu** [comment]
Hi :)

`/var/run/postgres` sounds like a path inside the postgres container and not firefly-pico. 
Can you share more logs about the error?

**Comment by Retr0astic** [comment]
/var/run/postgres is the path I've mounted my postgres socket file into the pico container so pico can access it, it has worked on all other versions so far. 

I can show the logs but all they contain is

"nc: bad address '/var/run/postgresql'
trying again in 10s"

**Comment by Retr0astic** [comment]
Also, downgrading to 1.4.1 lets pico connect to my db

**Comment by cioraneanu** [comment]
Ahhhh I see. You're Pico env looks something like this?
```
environment:
      - DB_HOST=/var/run/postgresql
```

The latest version has a wait for DB to be ready mechanism. I didn't think anyone would connect to it any other way 😁 
Pushed a possible fix in 9453a90. Give the `dev` docker tag a try and let me know if it's working for your config.

**Comment by Retr0astic** [comment]
Okay, I've tried the latest dev tag today and this issue is resolved. 

Thank you

**Comment by Retr0astic** [comment]
This issue is happening again in the latest dev version.

**Comment by cioraneanu** [comment]
I'm not sure why it would fail again, the only new commit after the attempted fix is this one 0eb14d7. Take a look at it and let me know if you find why it fails with your setup.

**Comment by Retr0astic** [comment]
Okay, i have figured out the issue:

I used to have /var/run/postgresql as my socket path. But after the latest main and dev branch update, just passing the folder is not enough. 

You have to pass the socket file itself, that is /var/run/postgresql/.s.PGSQL.5432.

**Comment by Retr0astic** [comment]
I'm not sure what changed that's causing this issue again, but I can confirm that going back to Version 1.4.1 Main branch lets me connect again to the postgres socket.

**Comment by cioraneanu** [comment]
That is expected, the wait for DB mechanism is not yet of the `main` branch. But it is necesary and it will be mergem into `main` soon. Currently if you have both Pico and Postgres into the same docker compose you're relying on luck for the DB container to start before the Pico container.

To fix this isssue you need to create a container from the latest docker `dev`tag, SSH into it, and change this file `vi /docker-entrypoint.d/start.sh`. You need to tweak the "Unix socket" part so that it properly works with your config. After each change restart the container and check the logs if it's working.

**Comment by Retr0astic** [comment]
I'm running this on podman, and making any changes to /docker-entrypoint.d/start.sh and then restarting reverts my changes to original.

**Comment by Retr0astic** [comment]
update on this issue,
Changing  
if [ -S "$DB_HOST" ]; then 

to 
if [ -S "/var/run/postgresql/.s.PGSQL.5432:" ]; then 

lets pico connect to the database succesfuly

**Comment by cioraneanu** [comment]
Ok... This means that if you set your `DB_HOST` variable to `/var/run/postgresql/.s.PGSQL.5432` everything should works as intented right? 😃

**Comment by Retr0astic** [comment]
Yes, after testing for a day, everything is working as intended.

**Comment by Retr0astic** [comment]
There has been a misunderstanding on my side:

Setting the variable to /var/run/postgresql/.s.PGSQL.5432 does not let the container start successfully.

I got it working by editing start.sh and removing $DB_Host and instead using /var/run/postgresql/.s.PGSQL.5432.
